pkg/container/service.go

```diff
@@ -576,19 +634,134 @@ func (s *service) GetContainerKMI(containerID string) (kmi.KMI, error) {
 
 func (s *service) getContainerKMI(containerID string) (kmi.KMI, error) {
 	// TODO: cache container KMIs
+	cKMI, err := s.getCKMI(containerID)
+	if err != nil {
+		return kmi.KMI{}, err
+	}
+
+	return kmi.KMI(cKMI.KMI), nil
+}
+
+func (s *service) getCKMI(containerID string) (CKMI, error) {
 	c := Container{}
 	err := s.db.First(&c, "container_id = ?", containerID)
 	if err != nil {
-		return kmi.KMI{}, err
+		return CKMI{}, err
 	}
 
 	cKMI := CKMI{}
 	err = s.db.First(&cKMI, "id = ?", c.KMIID)
 	if err != nil {
-		return kmi.KMI{}, err
+		return CKMI{}, err
+	}
+
+	return cKMI, nil
+}
+
+func (s *service) getContainerIP(refID uint, containerID string) string {
+	ipPath := path.Join(s.config.CustomerPath, fmt.Sprintf("%d", refID), containerID, ".ip")
+	ip, err := ioutil.ReadFile(ipPath)
+	if err != nil {
+		return ""
+	}
+
+	return fmt.Sprintf("%s", ip)
```

Comment:
No, casting a byte slice to string is only possible when the length is known. Which is not the case when doing `.ReadFile()`